src/odemis/acq/move.py

```diff
@@ -0,0 +1,314 @@
+# -*- coding: utf-8 -*-
+"""
+Created on 27 July 2020
+
+@author: Éric Piel, Bassim Lazem
+
+Copyright © 2020 Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the
+terms  of the GNU General Public License version 2 as published by the Free
+Software  Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY;  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR  PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+"""
+
+from __future__ import division
+
+import logging
+import threading
+from asyncio import CancelledError
+from concurrent.futures._base import CANCELLED, RUNNING, FINISHED
+
+import numpy as np
+
+from odemis import model, util
+from odemis.util import executeAsyncTask
+
+TIMEOUT_DURATION = 60
+LOADING, IMAGING = 0, 1
+ATOL_POS = 100e-6
+ATOL_PROGRESS = 1e-3
+
+
+def getLoadingProgress(current_pos, start_pos, end_pos):
+    """
+    Computes the position on the path between LOADING and IMAGING.
+    If it’s too far from the line between LOADING → IMAGING, then it’s considered out of the path.
+    :param current_pos: Current position of the stage
+    :param start_pos: Either LOADING or IMAGING position to start the movement from
+    :param end_pos: Either LOADING or IMAGING position to end the movement to
+    :return: Percentage of the progress, None if it's out of the path
+    """
+
+    def get_distance(start, end):
+        # Calculate the euclidean distance between two 3D points
+        sp = np.array([start['x'], start['y'], start['z']])
+        ep = np.array([end['x'], end['y'], end['z']])
+        squared_dist = np.sum((ep - sp) ** 2, axis=0)
+        dist = np.sqrt(squared_dist)
```

Comment:
I think `numpy.linalg.norm(ep-sp)` should do the same as these two lines.

https://numpy.org/doc/stable/reference/generated/numpy.linalg.norm.html

Reply:
or `scipy.spatial.distance.euclidean(ep, sp)`

https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.distance.euclidean.html

Reply:
To continue on alternatives, you could also write: `math.hypot(*(ep - sp))`.... but that only works on Python 3.8 with more than 2 dimensions... so not really an alternative ;-)